src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java

```diff
@@ -118,6 +120,27 @@ private OneTimePassword getOneTimePassword(OneTimePasswordDTO oneTimePasswordDTO
         return oneTimePassword;
     }
 
+    @Override
+    public void check(OneTimePasswordCheckDTO oneTimePasswordDTO) {
+
+        OneTimePassword passwordEntity = oneTimePasswordRepository.getOne(oneTimePasswordDTO.getId());
+
+        if (passwordEntity.getStateKey() != StateKey.ACTIVE
```

Comment:
Could we find a more elegant way to write this condition?

Reply:
dome